### PR TITLE
Simplified Pub sub Sample

### PR DESCRIPTION
**Pull request by indualagarsamy**
- Simple pub/sub sample containing 1 publisher endpoint and 1 subscriber endpoint. 
- Removed multiple inheritance event definitions. For a user trying to see how pub/sub works, the current event definitions were too confusing - IEvent, EventMessage and AnotherEventMessage! Instead the sample has 1 event called, "OrderReceived" defined, that users can more relate to. 
- Removed the subscription authorization code from the sample
- The readme for the sample is now modified to reflect these changes. 

cc / @Particular/nservicebus-maintainers 

@Particular/docs-maintainers - can you please review?

### CODE REVIEW

**Comment by SimonCropp** [comment]
should we rename the endpoints to reflect the order/shipping domain?

**Comment by SimonCropp** [comment]
mark md as reviewed

File: samples/pubsub/sample.md [hunk 1]
```diff
@@ -33,35 +33,33 @@ snippet:PublishLoop
...
 
- * The `Subscriber1` process handles and subscribes to both the `EventMessage` and `AnotherEventMessage` types.
- * The `Subscriber2` handles and subscribes to any message implementing the interface `IMyEvent`.
+ * The `Subscriber` process handles and subscribes to the `OrderReceived` type.
 
-The handlers in each project are in files that end in with the word `Handler` for example `EventMessageHandler.cs`. Since both the `EventMessage` and `AnotherEventMessage` classes in the `Shared` project implement the `IMyEvent` interface, when they are published both subscribers receive it. When the specific message types of `EventMessage` and `AnotherEventMessage` are published, only the handlers of that specific type in `Subscriber1` are invoked.
-
- * `Subscriber1` uses the default auto-subscription feature of the bus where the the bus automatically sends subscription messages to the configured publisher.
- * `Subscriber2` explicitly disables the auto-subscribe feature in the `Program.cs` file. The subscriptions are therefore done explicitly at startup.
+ * The handlers in each project are in files that end in with the word `Handler` for example `OrderReceivedHandler.cs`.
```
Comment (SimonCropp) [inline samples/pubsub/sample.md]: remove the extra newlines

**Comment by indualagarsamy** [comment]
@SimonCropp - I'm on the fence about the renaming the endpoints. For a person trying to look at how a publisher works, having an endpoint named "Publisher" is much easier. Sales or Order or Billing doesn't easily identify which endpoint is the publisher vs the subscriber. That's why I kept "Publisher" and "Subscriber". Thoughts?

**Comment by SimonCropp** [comment]
@indualagarsamy i am ok with that

**Pull request merged**